numpy/random/tests/test_extending.py

```diff
@@ -39,55 +43,117 @@
     # other fixes in the 0.29 series that are needed even for earlier
     # Python versions.
     # Note: keep in sync with the one in pyproject.toml
-    required_version = '0.29.30'
+    required_version = '0.29.35'
     if _pep440.parse(cython_version) < _pep440.Version(required_version):
         # too old or wrong cython, skip the test
         cython = None
 
 
-@pytest.mark.skipif(sys.version_info >= (3, 12),
-                    reason="numpy.distutils not supported anymore")
 @pytest.mark.skipif(IS_WASM, reason="Can't start subprocess")
 @pytest.mark.skipif(cython is None, reason="requires cython")
 @pytest.mark.slow
 def test_cython(tmp_path):
-    from numpy.distutils.misc_util import exec_mod_from_location
+    import glob
+    # build the examples in a temporary directory
     srcdir = os.path.join(os.path.dirname(__file__), '..')
     shutil.copytree(srcdir, tmp_path / 'random')
-    # build the examples and "install" them into a temporary directory
     build_dir = tmp_path / 'random' / '_examples' / 'cython'
-    subprocess.check_call([sys.executable, 'setup.py', 'build', 'install',
-                           '--prefix', str(tmp_path / 'installdir'),
-                           '--single-version-externally-managed',
-                           '--record', str(tmp_path/ 'tmp_install_log.txt'),
-                          ],
-                          cwd=str(build_dir),
-                      )
+    # We don't want a wheel build, so do the steps in a controlled way
+    # The meson.build file is not copied as part of the build, so generate it
```

Comment:
This should not be necessary, you can install any file you want/need. Can you commit a regular `meson.build` file in `random/_examples/cython/` and edit `random/meson.build` to install it rather than the `_examples/cython/setup.py` file that's installed now?